meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -31,15 +31,21 @@
 import org.apache.helix.metaclient.api.DirectChildSubscribeResult;
 import org.apache.helix.metaclient.api.MetaClientInterface;
 import org.apache.helix.metaclient.api.OpResult;
-import org.apache.helix.zookeeper.api.client.RealmAwareZkClient;
+import org.apache.helix.metaclient.impl.zk.factory.ZkMetaClientConfig;
+import org.apache.helix.zookeeper.impl.client.ZkClient;
+import org.apache.helix.zookeeper.zkclient.ZkConnection;
 
 
 public class ZkMetaClient implements MetaClientInterface {
 
-  private RealmAwareZkClient _zkClient;
-
-  public ZkMetaClient() {
+  private final ZkClient _zkClient;
 
+  public ZkMetaClient(ZkMetaClientConfig config) {
+    _zkClient =  new ZkClient(new ZkConnection(config.getConnectionAddress(),
+        (int) config.getSessionTimeoutInMillis()),
+        (int) config.getConnectionInitTimeoutInMillis(), -1 /*operationRetryTimeout*/,
+        config.getZkSerializer(), config.getMonitorType(), config.getMonitorKey(),
+        config.getMonitorInstanceName(), config.getMonitorRootPathOnly());
```

Comment:
Curious what's our preference here for instantiation? With constructor or builder class?

Reply:
The preferred method for Client instantiation is through Factory class, however, user could use constructor as well.  The config can only be created using Builder. 
(Same as Current FederatedZkClient.)